pkg/pods/pod.go

```diff
@@ -529,6 +529,12 @@ func (pod *Pod) setupConfig(manifest Manifest, launchables []launch.Launchable)
 	}
 
 	for _, launchable := range launchables {
+		// we need to remove any unset env vars
```

Comment:
If someone elides an environment variable from their pod manifest, it should not be present on the subsequent deploy (ie, we need to actively remove it from the env directory)